### DISCUSSION TITLE
Problems getting workflow to progress

### DISCUSSION THREAD

**Discussion by kjell-ohlsson** [argoproj/argo-workflows#6131] [Q&A] [answered]
Hi!
I've posted this into slack also.
I’ve installed Argo workflow on my win 10 pc using docker for windows, wsl2, Ubuntu app and KinD. All seems to be working, but when I kick off the example hello-World workflow, the wait container never terminates. It just keeps on writing log texts like ”time="2021-06-11T07:23:21.791Z" level=info msg="docker ps --all --no-trunc --format={{.Status}}|{{.Label \"io.kubernetes.container.name\"}}|{{.ID}}|{{.CreatedAt}} --filter=label=io.kubernetes.pod.namespace=argo --filter=label=io.kubernetes.pod.name=hello-world-rxrsn” to its log every second. Main container terminates as expected and prints out the expected whale to its log. Is “wait” waiting for something else to terminate? :-S
I have also noticed that if I run a multi-step workflow, the processing never leaves the very first step. I would assume there needs to be some kind of flagging from a terminating step to let the workflow control know that it's time for a subsequent step to start, or to tell wait container that all is done. Perhaps it's this signal that is lacking... Anyone experienced the same?

**Comment by kjell-ohlsson** [comment, marked as answer]
Answering my own question. It worked to switch workflow executor. I tested k8sapi and emissary. Both worked!

**Reply by shanebowyer** [reply]
Thank you this put me in the right direction.
`kubectl create configmap -n argo workflow-controller-configmap --from-literal=config="containerRuntimeExecutor: pns"`
[from this link](https://boxboat.com/2020/02/10/argo-workflows/)

**Reply by kjell-ohlsson** [reply]
Great that it helped you! Thanks for the boxboat-link. Will have a look at it! :-)
